scala-package/core/src/main/scala/org/apache/mxnet/NDArray.scala

```diff
@@ -262,16 +263,32 @@ object NDArray extends NDArrayBase {
     arr
   }
 
+  def full(shape: Shape, value: Double, ctx: Context): NDArray = {
+    val arr = empty(shape, ctx, DType.Float64)
+    arr.set(value)
+    arr
+  }
+
+  /**
+    * Create a new NDArray filled with given value, with specified shape.
+    * @param shape shape of the NDArray.
+    * @param value value to be filled with
+    */
+  def full(shape: Shape, value: Double): NDArray = {
+    full(shape, value, null)
+  }
+
+
   // Perform power operator
   def power(lhs: NDArray, rhs: NDArray): NDArray = {
     NDArray.genericNDArrayFunctionInvoke("_power", Seq(lhs, rhs))
   }
 
-  def power(lhs: NDArray, rhs: Float): NDArray = {
+  def power(lhs: NDArray, rhs: MX_PRIMITIVE_TYPE): NDArray = {
```

Comment:
curious what will be the input from the user perspective? Will this be `typeT` or simply the name as `MX_PRIMITIVE_TYPE`? Will users be able to get information what kind of types should be used here. If a new data type (for example `int8`) is not supported currently, will the user be able to get the correct error messages?

Reply:
The input from the user perspective will be either a ```Double``` or a ```Float```. 
The users in the scala docs would see a link to MX_PRIMITIVE_TYPE for support, which defines the primitives as Float or a Double.

If an incompatible datatype is passed, the IDE/compiler would throw an error because it would not be able to resolve the appropriate implicit conversion for MX_PRIMITIVE_TYPE.

Reply:
As discussed offline, I'll add comments on the methods changed in NDArray.